### ISSUE TITLE
3.2 sync relationships going reload after related parent creation

### ISSUE THREAD

**Issue by lifeart** [emberjs/data#5525]
Given:

JSONAPI adapter/serializer.
Prev version 2.18
Works ok in 2.18.3

`shouldBackgroundReloadRecord() => false`

```
user {
  item: belongsTo('item', { async: false }),
  book: belongsTo('book', { async: false }),
  toy: belongsTo('toy', { async: false });
}
```

going to create user with already loaded relationships, but, after save, ember-data trying to reload each relationship

**Comment by runspired** [comment]
@lifeart is this an issue in 3.1?

**Comment by lifeart** [comment]
@runspired, nope, can't reproduce it in 3.1.1

**Comment by runspired** [comment]
@lifeart I can't reproduce this in a test, could you give me more information about how you created the record, established the relationship, and what the response from the API looked like?

**Comment by lifeart** [comment]
follow up from slack

model looks like 
```js

export default DS.Model.extend({
    transplantationCenter: belongsTo('transplantation-center', {
        async: false
    }),
    clinicalData: belongsTo('clinical-info', {
        inverse: 'patient',
        async: false
    }),
    gender: belongsTo('gender', { async: false }),
    document: belongsTo('document', { async: false }),
    country: belongsTo('meta-place', { async: false, polymorphic: true }),
    status: belongsTo('patient-status', { async: false }),
    city: belongsTo('meta-place', { async: false, polymorphic: true }),
    consiliumList: belongsTo('patient/consilium', { async: false }),
    consiliums: hasMany('consilium', { async: false, inverse: 'patient' }),
    users: hasMany('basic-user', { async: false, polymorphic: true }),
    pickedDoctors: alias('users'),
    refusalList: hasMany('refuse'),
    phonesList: hasMany('patient/phone', { async: false, inverse: 'patient' }),
});
```


Issue exists only for new models (isNew = true)

for example, i do `Patient.setProperties({city: CityModel(42) })`, and do Patient.save(), I get one more request to `/city/42`


`POST (http 201)`
```js
 {"data":{"attributes":{"isSelected":false,"weight":123,"height":233,"firstName":"23232323","lastName":"22323","secondName":"232323","birthDate":"2018-07-16T00:00:00.000Z","status":0,"lastUpdatedBy":"Administrator","lastUpdatedDate":"2018-07-23T18:11:18.618Z","active":true},"relationships":{"transplantationCenter":{"data":{"type":"organization","id":"799"}},"gender":{"data":{"type":"gender","id":"2"}},"document":{"data":{"type":"documentCatalog","id":"1"}},"country":{"data":{"type":"metaPlace","id":"61373"}},"city":{"data":{"type":"metaPlace","id":"86189"}},"users":{"data":[{"type":"user","id":"5"}]}},"type":"patient"}} 
```

result


```js
{"data":{"id":"1319","type":"patient","attributes":{"lastName":"22323","note":null,"cityFullName":null,"retirementComment":null,"passportIssuer":null,"passportIssueDate":null,"secondName":"232323","height":233,"passportNumber":null,"lastUpdatedBy":"Administrator","address":null,"retirementDate":null,"weight":123,"active":true,"fullName":"22323 23232323 232323","birthDate":"2018-07-16T00:00:00.000Z","passportSeries":null,"firstName":"23232323","lastUpdatedDate":"2018-07-23T18:11:18.618Z","retirementId":null,"additionalAddress":null,"bodyArea":null,"region":"Прочие","shortName":"22323 2.2.","status":0},"relationships":{"meldPelds":{"data":[],"links":{"self":"/api/v2/patient/1319/relationships/meldPelds","related":"/api/v2/patient/1319/meldPelds"}},"country":{"data":{"id":"61373","type":"metaPlace"},"links":{"self":"/api/v2/patient/1319/relationships/country","related":"/api/v2/patient/1319/country"}},"gender":{"data":{"id":"2","type":"gender"},"links":{"self":"/api/v2/patient/1319/relationships/gender","related":"/api/v2/patient/1319/gender"}},"city":{"data":{"id":"86189","type":"metaPlace"},"links":{"self":"/api/v2/patient/1319/relationships/city","related":"/api/v2/patient/1319/city"}},"document":{"data":{"id":"1","type":"documentCatalog"},"links":{"self":"/api/v2/patient/1319/relationships/document","related":"/api/v2/patient/1319/document"}},"photo":{"data":null,"links":{"self":"/api/v2/patient/1319/relationships/photo","related":"/api/v2/patient/1319/photo"}},"transplantationCenter":{"data":{"id":"799","type":"organization"},"links":{"self":"/api/v2/patient/1319/relationships/transplantationCenter","related":"/api/v2/patient/1319/transplantationCenter"}},"patientPhonesList":{"data":[],"links":{"self":"/api/v2/patient/1319/relationships/patientPhonesList","related":"/api/v2/patient/1319/patientPhonesList"}},"users":{"data":[{"id":"5","type":"user"}],"links":{"self":"/api/v2/patient/1319/relationships/users","related":"/api/v2/patient/1319/users"}},"clinicalData":{"links":{"self":"/api/v2/patient/1319/relationships/clinicalData","related":"/api/v2/patient/1319/clinicalData"}},"consiliumList":{"data":[],"links":{"self":"/api/v2/patient/1319/relationships/consiliumList","related":"/api/v2/patient/1319/consiliumList"}},"smallPhoto":{"data":null,"links":{"self":"/api/v2/patient/1319/relationships/smallPhoto","related":"/api/v2/patient/1319/smallPhoto"}},"refusalList":{"links":{"self":"/api/v2/patient/1319/relationships/refusalList","related":"/api/v2/patient/1319/refusalList"}}},"links":{"self":"/api/v2/patient/1319"}}} 
```

and after this, I have "reload" requests for country, transplantationCenter, document and gender
and `reaload` requests use links from server relationships links, for example - `GET /api/v2/api/v2/patient/1319/city`,
GET `api/v2/patient/1319/country`
looks like ember-data trying to reload items, using `related` links, but, this relationships marked as `async: false` and already loaded

**Comment by runspired** [comment]
Traced this down to observer usage (observers are bad, plz avoid), which would trigger in an intermediate unsettled state while the relationship was updating. The fix moves the notification for observers to after the state is settled.

**Comment by runspired** [comment]
This was fixed in #5537 but I would like to add additional test coverage, leaving this open until such time.
